File: CS-Examples/CS-Examples/24_Workbook/SetMargins.cs
using System;
using System.Windows.Forms;
using Spire.Xls;

namespace SetMargins
{

	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, EventArgs e)
		{
            //Create a workbook
			Workbook workbook = new Workbook();

            //Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\WorksheetSample1.xlsx");

            //Get the first worksheet
            Worksheet sheet = workbook.Worksheets[0];

            //Set margins for top, bottom, left and right, here the unit of measure is Inch
            sheet.PageSetup.TopMargin = 0.3;
            sheet.PageSetup.BottomMargin = 1;
            sheet.PageSetup.LeftMargin = 0.2;
            sheet.PageSetup.RightMargin = 1;
            //Set the header margin and footer margin
            sheet.PageSetup.HeaderMarginInch = 0.1;
            sheet.PageSetup.FooterMarginInch = 0.5;

            //Save the document
            string output = "SetMargins.xlsx";
			workbook.SaveToFile(output, ExcelVersion.Version2013);

            //Launch the Excel file
			ExcelDocViewer(output);
		}
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}

File: VB-Examples/24_Workbook/SetMargins.vb
Imports Spire.Xls

Namespace SetMargins

	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the workbook from a file using a relative path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\WorksheetSample1.xlsx")

            ' Access the first worksheet in the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Set the top, bottom, left, and right margins of the worksheet's page setup
            sheet.PageSetup.TopMargin = 0.3
            sheet.PageSetup.BottomMargin = 1
            sheet.PageSetup.LeftMargin = 0.2
            sheet.PageSetup.RightMargin = 1

            ' Set the header and footer margins of the worksheet's page setup
            sheet.PageSetup.HeaderMarginInch = 0.1
            sheet.PageSetup.FooterMarginInch = 0.5

            ' Specify the output file name for saving the modified workbook
            Dim output As String = "SetMargins.xlsx"

            ' Save the workbook to a file with the specified output name and Excel version
            workbook.SaveToFile(output, ExcelVersion.Version2013)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the Excel file
            ExcelDocViewer(output)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
